### PR TITLE
Mock-up for #24622

### PR DESCRIPTION
**Pull request by MoritzS**
Mock-up for suggestion in [Ticket 24622](https://code.djangoproject.com/ticket/24622).
Not intended to be merged.

### CODE REVIEW

**Comment by charettes** [comment]
Looks promising. If we adopt this approach I think we should be fully backend agnostic and set `template_class` on both `backends.django.DjangoTemplates` and `backends.dummy.TemplateStrings` and remove the `backends.django.Template` special casing.

File: django/test/utils.py [hunk 1]
```diff
@@ -127,6 +145,11 @@ def teardown_test_environment():
     Template._render = Template._original_render
     del Template._original_render
 
+    for engine in engines:
+        engine_cls = engines[engine]
+        if hasattr(engine_cls, '_original_render'):
+            engine_cls.render = engine_cls._original_render
```
Comment (charettes) [inline django/test/utils.py]: These checks should be run against `engine_cls.template_class` and not `engine_cls`.

Comment (MoritzS) [inline django/test/utils.py]: Whoops, yeah you're right.

**Comment by charettes** [comment]
I guess we'll have to document that `template_rendered` is now sent by backend specific `Template` classes and not the base `Template` class now.

**Comment by MoritzS** [comment]
Closing this. More details on the ticket.

**Pull request closed without merging**